### PR TITLE
NonNull annotations, sup-lib 23.4.0 & Issue #85

### PR DESCRIPTION
**Pull request by tomaszrykala**

### CODE REVIEW

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskContract.java [hunk 1]
```diff
@@ -41,7 +41,9 @@
 
         void createTask(String title, String description);
 
-        void updateTask( String title, String description);
+        void updateTask(String title, String description);
+
+        void onTaskFabClick(String title, String description);
```
Comment (JoseAlcerreca) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskContract.java]: This name in the contract is referring to a detail in the view. The presenter shouldn't care about it being a FAB or a menu item. saveTask(...) would be better.

**Comment by JoseAlcerreca** [comment]
Minor comment, otherwise looks good, thanks! 

I'll merge this into the rest of the branches. That's going to be fun.

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskContract.java (hunk 1 above)
Comment (tomaszrykala) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskContract.java]: Will update, thanks.

**Comment by tomaszrykala** [comment]
@JoseAlcerreca Happy to help with merging :-)

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskContract.java [hunk 2]
```diff
@@ -41,7 +41,9 @@
 
         void createTask(String title, String description);
 
-        void updateTask( String title, String description);
+        void updateTask(String title, String description);
```
Comment (JoseAlcerreca) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskContract.java]: sorry I missed this, createTask and updateTask need to be removed and made private in the concrete class.

Comment (tomaszrykala) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskContract.java]: Will correct, thanks. Had a feeling about this :-)

**Comment by JoseAlcerreca** [comment]
Thank you Tomasz!

**Comment by JoseAlcerreca** [comment]
All active branches merged.

**Pull request merged**